### PR TITLE
fix: Add missing logs at startup

### LINKED ISSUE #2308: Missing info message on start
We use to have
```
2024-01-29T14:17:19.131-0500, INFO, cli, Starting DefraDB service...
2024-01-29T14:17:19.132-0500, INFO, cli, Opening badger store, {"Path": "/Users/fredcarle/.defradb/data"}
2024-01-29T14:17:19.146-0500, INFO, cli, Starting P2P node, {"P2P addresses": ["/ip4/0.0.0.0/tcp/9171", "/ip4/127.0.0.1/tcp/9171"]}
2024-01-29T14:17:19.155-0500, INFO, net, Created LibP2P host, {"PeerId": "12D3KooWJXPEb8uTgLjYQdSySxmSB1SWuNMpn7jMuqgTSsd7SEky", "Address": ["/ip4/0.0.0.0/tcp/9171", "/ip4/127.0.0.1/tcp/9171"]}
2024-01-29T14:17:19.157-0500, INFO, net, Starting internal broadcaster for pubsub network
2024-01-29T14:17:19.160-0500, INFO, cli, Providing HTTP API at http://127.0.0.1:9181.
```

But we now only have
```
2024-02-12T10:35:42.722-0500, INFO, net, Created LibP2P host, {"PeerId": "12D3KooWMLTavsJBsVz7AQw5aiJQAsf8ew1WyJayX2vPhCWyk3xL", "Address": ["/ip4/127.0.0.1/tcp/9171"]}
2024-02-12T10:35:42.724-0500, INFO, cli, Starting DefraDB
2024-02-12T10:35:42.724-0500, INFO, net, Starting internal broadcaster for pubsub network
```

We should add back the store info and the HTTP API endpoint addess.

### LINKED ISSUE #2309: Listener error logged on server close
We use to have
```
2024-01-29T15:14:16.501-0500, INFO, cli, Received interrupt; closing database...
2024-01-29T15:14:16.502-0500, INFO, db, Closing DefraDB process...
2024-01-29T15:14:16.504-0500, INFO, db, Successfully closed running process
```
and now we have
```
2024-02-12T14:37:30.282-0500, INFO, cli, Received interrupt; shutting down...
2024-02-12T14:37:30.282-0500, ERROR, node, HTTP server stopped, {"Error": "http: Server closed"}
2024-02-12T14:37:30.283-0500, INFO, db, Closing DefraDB process...
2024-02-12T14:37:30.293-0500, INFO, db, Successfully closed running process
```

We shouldn't log a `Server closed` error from the listener.

### PR DESCRIPTION
**Pull request by fredcarle**
## Relevant issue(s)

Resolves #2308 
Resolves #2309

## Description

This PR re-adds startup logs showing the P2P addresses and HTTP endpoint of the node.

## Tasks

- [x] I made sure the code is well commented, particularly hard-to-understand areas.
- [x] I made sure the repository-held documentation is changed accordingly.
- [x] I made sure the pull request title adheres to the conventional commit style (the subset used in the project can be found in [tools/configs/chglog/config.yml](tools/configs/chglog/config.yml)).
- [x] I made sure to discuss its limitations such as threats to validity, vulnerability to mistake and misuse, robustness to invalidation of assumptions, resource requirements, ...

## How has this been tested?

manually tested in the CLI

Specify the platform(s) on which this was tested:
- MacOS

### CODE REVIEW

**Review by AndrewSisley — APPROVED** [review]
LGTM, assuming you fix the lint error before merge :)

**Review by shahzadlone — APPROVED** [review]
Cheers

**Pull request merged**
